core/lowering/passes/remove_dropout.cpp

```diff
@@ -1,92 +1,55 @@
-  remove_feature_dropout_inplace_pattern.runOnGraph(graph);
-
-  // remove feature_alpha_dropout
-  std::string feature_alpha_dropout_pattern = R"IR(
-        graph(%input, %4, %5):
-            %6 = aten::feature_alpha_dropout(%input, %4, %5)
-            return (%6))IR";
-  std::string no_feature_alpha_dropout_pattern = R"IR(
-        graph(%input, %4, %5):
-            return (%input))IR";
-
-  torch::jit::SubgraphRewriter remove_feature_alpha_dropout_pattern;
-  remove_feature_alpha_dropout_pattern.RegisterRewritePattern(
-      feature_alpha_dropout_pattern, no_feature_alpha_dropout_pattern);
-  remove_feature_alpha_dropout_pattern.runOnGraph(graph);
-
-  // remove feature_alpha_dropout inplace
-  std::string feature_alpha_dropout_inplace_pattern = R"IR(
-        graph(%input, %4, %5):
-            %6 = aten::feature_alpha_dropout_(%input, %4, %5)
-            return (%6))IR";
-  std::string no_feature_alpha_dropout_inplace_pattern = R"IR(
-        graph(%input, %4, %5):
-            return (%input))IR";
-
-  torch::jit::SubgraphRewriter remove_feature_alpha_dropout_inplace_pattern;
-  remove_feature_alpha_dropout_inplace_pattern.RegisterRewritePattern(
-      feature_alpha_dropout_inplace_pattern, no_feature_alpha_dropout_inplace_pattern);
-  remove_feature_alpha_dropout_inplace_pattern.runOnGraph(graph);
+// Schemas for dropout variants
+const std::unordered_set<c10::Symbol> DropoutSchemas = {
+    c10::Symbol::fromQualString("aten::dropout"),
+    c10::Symbol::fromQualString("aten::dropout_"),
+    c10::Symbol::fromQualString("aten::feature_dropout"),
+    c10::Symbol::fromQualString("aten::feature_dropout_"),
+    c10::Symbol::fromQualString("aten::feature_alpha_dropout"),
+    c10::Symbol::fromQualString("aten::feature_alpha_dropout_"),
+};
+
+void removeDropoutInBlock(torch::jit::Block* block) {
+  /*
+  Function adapted from:
+  torch/csrc/jit/passes/remove_dropout.cpp
+
+  Modified for conciseness, documentation, and allowing new variants of dropout operators to be quickly added
+  */
+  std::vector<torch::jit::Node*> dropout_nodes_to_remove;
+
+  for (auto node : block->nodes()) {
+    // Remove dropout for each member block within a node
+    for (auto block : node->blocks()) {
+      removeDropoutInBlock(block);
+    }
+
+    // For each node having a dropout-variant Schema, remove the node
+    if (DropoutSchemas.find(node->kind()) != DropoutSchemas.end()) {
+      // Extract input and output tensors of dropout operator
+      auto input_value = node->inputs()[0];
+      auto output_value = node->outputs()[0];
+
+      output_value->replaceAllUsesWith(input_value);
+      dropout_nodes_to_remove.push_back(node);
+    }
+  }
+
+  // Delete dropout nodes
+  for (auto del_node : dropout_nodes_to_remove) {
+    del_node->destroy();
+  }
+}
 
+void RemoveDropout(std::shared_ptr<torch::jit::Graph>& graph) {
+  // Remove all instances of dropout variants from graph
+  removeDropoutInBlock(graph->block());
```

Comment:
Would we want to run dead code elimination here just to make sure the graph is clean for the next pass?

Reply:
Thanks for the suggestion - that sounds good, I have now added the dead code elimination